code/ui/generalhud/menu/Menu.Settings.Generator.cs

```diff
@@ -0,0 +1,146 @@
+
+                                                Dropdown dropdownSelection = uiPanel.Add.Dropdown(propertyName.ToLower());
+
+                                                foreach (PropertyInfo possibleDropdownPropertyInfo in propertyInfo.PropertyType.GetProperties())
+                                                {
+                                                    foreach (object possibleDropdownAttribute in possibleDropdownPropertyInfo.GetCustomAttributes())
+                                                    {
+                                                        if (possibleDropdownAttribute is DropdownOptionsAttribute dropdownOptionsAttribute && dropdownOptionsAttribute.DropdownSetting.Equals(subPropertyInfo.Name))
+                                                        {
+                                                            foreach (KeyValuePair<string, object> keyValuePair in Utils.GetPropertyValue<Dictionary<string, object>>(propertyObject, possibleDropdownPropertyInfo.Name))
+                                                            {
+                                                                dropdownSelection.AddOption(keyValuePair.Key, keyValuePair.Value);
+                                                            }
+
+                                                            dropdownSelection.OnSelectOption = (option) =>
+                                                            {
+                                                                Utils.SetPropertyValue(propertyObject, propertyName, (string) option.Data);
+
+                                                                if (settings is ServerSettings serverSettings)
+                                                                {
+                                                                    SettingFunctions.SendSettingsToServer(serverSettings);
+                                                                }
+                                                                else
+                                                                {
+                                                                    Event.Run(TTTEvent.Settings.Change);
+                                                                }
+                                                            };
+                                                        }
+                                                    }
+                                                }
+
+                                                dropdownSelection.SelectByData(Utils.GetPropertyValue<string>(propertyObject, propertyName));
+                                            }
+
+                                            break;
+                                    }
+
+                                    break;
+                                }
+                            }
+                        }
+                    }, categoryName.ToLower());
```

Comment:
We should break up each of these cases into a separate method, it's a bit of a huge method.